lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -151,7 +151,7 @@ public function connect($connectionName = null)
         // If we have a connection open, and this is not an explicit connection
         // change request, then abort right here, because we are already done.
         // This prevents writes to the slave in case of "keepSlave" option enabled.
-        if ($this->_conn && !$requestedConnectionChange) {
+        if (isset($this->_conn) && $this->_conn && !$requestedConnectionChange) {
```

Comment:
this change is wrong IMO. The property exists in the class (we already fixed the bug unsetting the property rather than the value previously), and the case of null is already covered by the condition before the patch